visual-recognition/v4.ts

```diff
@@ -99,64 +107,47 @@ class VisualRecognitionV4 extends BaseService {
    * You can also include images with the **images_file** parameter.
    * @param {number} [params.threshold] - The minimum score a feature must have to be returned.
    * @param {OutgoingHttpHeaders} [params.headers] - Custom request headers
-   * @param {Function} [callback] - The callback that handles the response
    * @returns {Promise<VisualRecognitionV4.Response<VisualRecognitionV4.AnalyzeResponse>>}
    */
-  public analyze(params: VisualRecognitionV4.AnalyzeParams, callback?: VisualRecognitionV4.Callback<VisualRecognitionV4.AnalyzeResponse>): Promise<VisualRecognitionV4.Response<VisualRecognitionV4.AnalyzeResponse>> {
-    const _params = extend({}, params);
-    const _callback = callback;
+  public analyze(params: VisualRecognitionV4.AnalyzeParams): Promise<VisualRecognitionV4.Response<VisualRecognitionV4.AnalyzeResponse>> {
+    const _params = Object.assign({}, params);
     const requiredParams = ['collectionIds', 'features'];
 
-    return new Promise((resolve, reject) => {
-      const missingParams = getMissingParams(_params, requiredParams);
-      if (missingParams) {
-        if (_callback) {
-          _callback(missingParams);
-          return resolve();
-        }
-        return reject(missingParams);
-      }
-
-      const formData = {
-        'collection_ids': Array.isArray(_params.collectionIds) ? _params.collectionIds.join(',') : _params.collectionIds,
-        'features': Array.isArray(_params.features) ? _params.features.join(',') : _params.features,
```

Comment:
See my comment on the v3 SDK

Reply:
funny thing about v4 - it supports these as both comma separated strings and separate form parts. I think they are generated as separate form parts so no handwritten change needed anymore for this.